pontoon/sync/models.py

```diff
@@ -45,6 +50,37 @@ def finished(self):
     def get_absolute_url(self):
         return reverse('pontoon.sync.logs.details', kwargs={'sync_log_pk': self.pk})
 
+    def fix_stats(self):
+        """
+        Recalculate any broken stats when sync task is finished. This is a
+        temporary fix for bug 1310668.
+        """
+
+        if not self.finished:
+            return
+
+        # total_strings missmatch between TranslatedResource & Resource
+        for t in TranslatedResource.objects.exclude(total_strings=F('resource__total_strings')):
+            t.total_strings = t.resource.total_strings
+            t.save()
```

Comment:
`F()` is allowed in `update()`, but [not with joined fields](https://code.djangoproject.com/ticket/14104). :/

I've ammended the commit.